Numerology/Numerology/clientapp/src/components/dictionaries/ProductsFormDialog.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { useForm } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import { toast } from 'react-toastify'
+import TextField from '@mui/material/TextField'
+import Stack from '@mui/material/Stack'
+import Dialog from '@mui/material/Dialog'
+import DialogContent from '@mui/material/DialogContent'
+import DialogTitleReusable from '../common/DialogTitleReusable'
+import DialogActionsReusable from '../common/DialogActionsReusable'
+import SwitchReusable from '../common/SwitchReusable'
+import ProductsService from '../../services/ProductsService'
+
+const initialState = {
+  id: 0,
+  name: '',
+  description: '',
+  price: null,
+  favourite: false,
+}
+
+const ProductsFormDialog = ({ productEditData, closeDialog, newProduct }) => {
+  const { t } = useTranslation()
+  const { register, handleSubmit, reset, control } = useForm({
+    defaultValues: {
+      ...(typeof productEditData !== null ? productEditData : initialState),
+    },
+  })
+
+  const onSubmit = async (data) => {
+    const productData = JSON.stringify({
+      id: data.id,
+      name: data.name,
+      description: data.description,
+      price: data.price,
+      favourite: data.favourite,
+    })
+    console.log(productData)
+    const res = await ProductsService.AddOrUpdateProduct(productData)
+    if (res.isError) {
+      toast.error(t('ErrorUpdating'))
+      return
+    }
+    toast.success(t('Added'))
+    closeDialog(true)
+    newProduct()
+  }
+
+  const deleteProduct = async () => {
+    const res = await ProductsService.DeleteProduct(productEditData.id)
+    if (res.isError) {
+      toast.error(t('ErrorDeleting'))
+      return
+    }
+    toast.success(t('Deleted'))
+    closeDialog(true)
+  }
+
+  const handleClear = () => {
+    reset(initialState)
+  }
+
+  const handleKeyPress = (event) => {
+    if (event.key === 'Enter') {
+      handleSubmit(onSubmit)()
+    }
+  }
+
+  return (
+    <Dialog open={true}>
+      <DialogTitleReusable title={'Product'} setOpenDialog={closeDialog} />
+      <DialogContent>
+        <Stack spacing={3} onSubmit={handleSubmit(onSubmit)} className='form-title'>
+          <TextField variant='standard' {...register('name')} label={t('Designation')} autoFocus onKeyPress={handleKeyPress} required />
```

Comment:
```suggestion
          <TextField variant='standard' {...register('name')} label={t('Designation')} autoFocus required />
```
To można zrobić lepiej

Reply:
To musisz mi wyjaśnić co masz na myśli.